cli/src/tasks.rs

```diff
@@ -73,16 +75,33 @@ impl TaskAction {
                 max_batch_size,
                 expiration,
                 hpke_config_id,
-                buckets,
+                categorical_buckets,
+                continuous_buckets,
                 length,
                 bits,
                 time_precision,
             } => {
                 let vdaf = match vdaf {
                     VdafName::Count => Vdaf::Count,
-                    VdafName::Histogram => Vdaf::Histogram {
-                        buckets: buckets.unwrap(),
-                    },
+                    VdafName::Histogram => {
+                        match (length, categorical_buckets, continuous_buckets) {
+                            (Some(length), None, None) => {
+                                Vdaf::Histogram(Histogram::Length { length })
+                            }
+                            (None, Some(buckets), None) => {
+                                Vdaf::Histogram(Histogram::Categorical { buckets })
```

Comment:
I'm not following the inclusion of Histogram::Categorical, as that's not a representation captured in either DAP specs (IIRC?). Was there a discussion somewhere that captured the context for this?

Reply:
Yes, although I'm not sure exactly where. The idea is that although dap is unconcerned with buckets, the divviup client can present a façade that transforms a raw measurement into a bucket id, preventing a category of error where there's a mismatch in bucket order. This is optional, and opaque buckets are also supported

Reply:
To clarify, we're not talking about the pure dap client, but the client in https://github.com/divviup/divviup-ts/pull/375 that presents a layer of custom divviup stuff on top of dap

Reply:
Found it! https://isrg.slack.com/archives/C0167LT4C73/p1692305959215659